src/test/util_tests.cpp

```diff
@@ -2078,6 +2078,55 @@ BOOST_AUTO_TEST_CASE(test_spanparsing)
     BOOST_CHECK_EQUAL(SpanToStr(results[3]), "");
 }
 
+BOOST_AUTO_TEST_CASE(test_SplitString)
+{
+    // Empty string.
+    {
+        std::vector<std::string> result = SplitString("", '-');
+        BOOST_CHECK_EQUAL(result.size(), 1);
+        BOOST_CHECK_EQUAL(result[0], "");
+    }
+
+    // Empty items.
+    {
+        std::vector<std::string> result = SplitString("-", '-');
+        BOOST_CHECK_EQUAL(result.size(), 2);
+        BOOST_CHECK_EQUAL(result[0], "");
+        BOOST_CHECK_EQUAL(result[1], "");
+    }
+
+    // More empty items.
+    {
+        std::vector<std::string> result = SplitString("--", '-');
+        BOOST_CHECK_EQUAL(result.size(), 3);
+        BOOST_CHECK_EQUAL(result[0], "");
+        BOOST_CHECK_EQUAL(result[1], "");
+        BOOST_CHECK_EQUAL(result[2], "");
+    }
+
+    // Separator is not present.
+    {
+        std::vector<std::string> result = SplitString("abc", '-');
+        BOOST_CHECK_EQUAL(result.size(), 1);
+        BOOST_CHECK_EQUAL(result[0], "abc");
+    }
+
+    // Basic behavior.
+    {
+        std::vector<std::string> result = SplitString("a-b", '-');
+        BOOST_CHECK_EQUAL(result.size(), 2);
+        BOOST_CHECK_EQUAL(result[0], "a");
+        BOOST_CHECK_EQUAL(result[1], "b");
+    }
+
+    // Case-sensitivity of the separator.
+    {
+        std::vector<std::string> result = SplitString("AAA", 'a');
+        BOOST_CHECK_EQUAL(result.size(), 1);
+        BOOST_CHECK_EQUAL(result[0], "AAA");
```

Comment:
weird issue on macos:

```
test/util_tests.cpp:2087: error: in "util_tests/test_SplitString": check result[0] == "" has failed [ != ]
test/util_tests.cpp:2095: error: in "util_tests/test_SplitString": check result[1] == "" has failed [ != ]
test/util_tests.cpp:2104: error: in "util_tests/test_SplitString": check result[2] == "" has failed [ != ]
test/util_tests.cpp:2111: error: in "util_tests/test_SplitString": check result[0] == "abc" has failed [abc != abc]
test/util_tests.cpp:2119: error: in "util_tests/test_SplitString": check result[1] == "b" has failed [b != b]
test/util_tests.cpp:2126: error: in "util_tests/test_SplitString": check result[0] == "AAA" has failed [AAA != AAA]

Reply:
I think the issue here is that the conversion of the string literals to `Span<const char>` works so that the `\0` is part of the span. So for `""` the size of the span is actually 1 with content "\0", while for `std::string` the size would be 0.

Reply:
Ah, good point, that also explains why only the last element of each result vector failed the test. Thanks, fixed.